File: subroutines/define-and-call/java/run_quiz.java
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science,
part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

import java.util.Scanner;

class DefineAndCall
{
    // The main method is the entry point for all Java programs
    public static void main(String[] args)
    {
        // Use Scanner for reading user input from the console
        Scanner scanner = new Scanner(System.in);

        System.out.println("What is your name? ");
        String name = scanner.nextLine();
        System.out.println("Greetings " + name);

        runQuiz();
        System.out.println("End of the quiz");
    }


    // Simulates a quiz
    public static void runQuiz()
    {
        // Use Scanner for reading user input from the console
        Scanner scanner = new Scanner(System.in);

        System.out.println("What is the capital city of Botswana?");
        String answer = scanner.nextLine();

        if ("Gaborone".equals(answer)) {
            System.out.println("Well done!");
        }
        else {
            System.out.println("Sorry! That is the wrong answer");
        }
    }


}

File: subroutines/define-and-call/python/run_quiz.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0


def run_quiz():
    """A very simple quiz"""
    answer = input("What is the capital city of Botswana? ")
    if answer == "Gaborone":
        print("Well done!")
    else:
        print("Sorry! That is the wrong answer")


def main():
    name = input("What is your name? ")
    print(f"Welcome {name}")
    run_quiz() 
    print("End of the quiz")   
    

# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == "__main__":
    main()
